activestorage/app/models/active_storage/variation.rb

```diff
@@ -59,14 +59,14 @@ def transform(file, &block)
 
   def format
     transformations.fetch(:format, :png).tap do |format|
-      if MiniMime.lookup_by_extension(format.to_s).nil?
+      if Marcel::Magic.by_extension(format.to_s).nil?
         raise ArgumentError, "Invalid variant format (#{format.inspect})"
       end
     end
   end
 
   def content_type
-    MiniMime.lookup_by_extension(format.to_s).content_type
+    Marcel::MimeType.for(extension: format.to_s)
```

Comment:
Current code assumes `content_type`  is always returned, so using `Marcel::MimeType`  which also always returns a value.